files/mcloud/node_modules/@appium/base-driver/build/lib/protocol/protocol.js

```diff
@@ -393,128 +393,126 @@ function buildHandler(app, method, path, spec, driver, isSessCmd) {
         }
         catch (err) {
 
-            async function updateDeviceStatus(token, udid, note) {
+            // token: string - auth token from stf 
+            // udid: string -  unique id of device 
+            // note: string - comment with error for device which would shows in stf
+            // isDeviceStatusChange: boolean - if true device status will change
+            // status: string - new status for device 
+            async function updateDevice(token, udid, note, isDeviceStatusChange, status) {
```

Comment:
we should combine `isDeviceStatusChange, status` into the status only.
if it is null/nil, then update only comment )